packages/graphql-docs-generator/src/generator/getFields.ts

```diff
@@ -40,6 +40,17 @@ export default function getFields(
       return getFields(subField, schema, adjustDepth(subField, depth), options);
     })
     .filter(f => f);
+
+  // add __typename to selection set.
+  // getFields() does not include __typename because __typename is implicitly included on all object types.
```

Comment:
This assumption might not be valid when we accept any valid GraphQL introspection schema (not generated using our transformers) from the customer that could have this field. Would be more inclined to have a check if this exists and then add it.